src/common/util/date.ts

```diff
@@ -10,20 +10,34 @@ export const dateFormatter = (value: Date | string | number, locale?: Locale) =>
   return `${exact} (${relative})`
 }
 
-export const formatDateString = (dateString: string | Date) => {
-  const date = new Date(dateString)
-  const day = date.getDate().toString().padStart(2, '0')
-  const month = (date.getMonth() + 1).toString().padStart(2, '0')
-  const year = date.getFullYear()
-
-  return `${day}.${month}.${year}`
+export const formatDateString = (dateString: string | Date, locale?: string | undefined) => {
+  return format(new Date(dateString), getDateFormat(locale ?? ''))
 }
 
 export const getRelativeDate = (value: Date | string, locale?: Locale) => {
   const date = new Date(value)
   return formatRelative(date, new Date(), { locale })
 }
 
+/**
+ * Value format for storing and passing date strings. Should be used whenever we pass date strings between components.
+ */
+export const DATE_VALUE_FORMAT = 'yyyy-MM-dd'
+
+/**
+ * Utility to return localized date format string. Should be used whenever dates are displayed to end users.
+ * @param language Language to get the corresponding date format
+ * @returns Format string
+ */
+export const getDateFormat = (language: string) => {
+  const defaultFormat = 'dd.MM.yyyy'
+  const languageDateFormats: { [language: string]: string } = {
+    bg: defaultFormat,
+    'en-US': 'MM/dd/yyyy',
+  }
+  return languageDateFormats[language] ?? defaultFormat
+}
```

Comment:
@batebobo We might also use the standard js libs to accomplish this functionality:

```
new Intl.DateTimeFormat(['bg', 'bg']).format(date)
```

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat

Reply:
Thank you! I spent a while looking for generic solutions for proper formatting, but somehow managed to miss that :smile: I'll change it and let you know :)

Reply:
Hi again! I removed the hardcoded date formats from the function in favor of a more generic solution using `formatToParts` and extracting the relevant fields. The reason I need the format and not the formatted date string is because the MUI datepicker expects a format string. The reason I need a format string is that the `date-fns` localization adapter produces `dd/MM/yyyy` for `bg` locale 😕  Let me know how this looks 🙏